src/surge-xt/gui/overlays/FilterAnalysis.cpp

```diff
@@ -238,8 +244,41 @@ void FilterAnalysis::paint(juce::Graphics &g)
             g.setColour(skin->getColor(Colors::MSEGEditor::Axis::SecondaryText));
             g.drawFittedText(dbString, labelRect, juce::Justification::right, 1);
         }
-    }
 
+        // draws the ruler to show the position of cutoff frequency and resonance
+        {
+            //    Tunings::Tuning tuning{};
+            //    double freq = tuning.frequencyForMidiNote(evaluator->cutoff + 69.0); // might be, but not gradual
+            double freq = std::pow(2, (evaluator->cutoff) / 12) * 440.0;
+            const auto xPos = freqToX(freq, width);
+
+            int yDraw = height - evaluator->resonance * height;
+
+            g.setColour(skin->getColor(Colors::MSEGEditor::CurveHighlight));
+            float r =  width * 0.008f;;
+            g.fillEllipse(xPos - r / 2.f, yDraw - r / 2.f, r, r);
+
+
+            // Draw vertical line
+            juce::Line line{juce::Point{xPos, 0.f}, juce::Point{xPos, (float)height}};
+            g.setColour(skin->getColor(Colors::MSEGEditor::Axis::Line));
+            g.drawLine(line);
```

Comment:
Nit: g.drawVerticalLine tends to be a little bit faster / more efficient on the backend (according to juce docs anyway).

Reply:
Done